### ISSUE TITLE
support tags in IRCv3

### ISSUE THREAD

**Issue by jaraco** [jaraco/irc#62]
http://ircv3.net/specs/core/message-tags-3.2.html

twitch chat already using it, but for now i parse them in `on_all_raw_messages`

---
- Bitbucket: https://bitbucket.org/jaraco/irc/issue/62
- Originally reported by: Mihail Latyshov
- Originally created at: 2015-05-01T12:03:09.751

**Comment by jaraco** [comment]
Sounds good. Would you be willing to put together a pull request?

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
i looked into it before, but its too complicated for me

---

Original comment by: Mihail Latyshov

**Comment by jaraco** [comment]
I would be happy to help, but I don't have the time to be able to design and implement this feature from scratch.

Let's start from the beginning; can you document what sort of user experience you'd like to see from the library? For example, if the feature were implemented how would you propose that twitch solicit the tags from the ServerConnection? Would tags be associated with any Event?

---

Original comment by: Jason R. Coombs

**Comment by jaraco** [comment]
> can you document what sort of user experience you'd like to see from the library?

nothing fancy, just `tags` inside `event`

i would use it like this

``` python3
def on_pubmsg(self, connection, event):
    logging.info(event.tags)
```

`tags` will be just dict

> Would tags be associated with any Event?

if there are `tags`, `event` will have `tags` inside, otherwise just `None`

but as i said, i already use them via `on_all_raw_messages`, so this issue is just a proposal, nothing else

---

Original comment by: Mihail Latyshov

**Comment by jaraco** [comment]
I added a Pull request to handle the IRCv3 tags.

---

Original comment by: [kimen91](http://bitbucket.org/kimen91)

**Comment by jaraco** [comment]
Pull Request #34 has been released as 12.2. Please test and re-open if that doesn't provide the solution. Huge thanks to @kimen91.

---

Original comment by: Jason R. Coombs
